src/app/clusters/bindings/BindingManager.cpp

```diff
@@ -101,31 +101,28 @@ CHIP_ERROR BindingManager::EstablishConnection(const ScopedNodeId & nodeId)
     VerifyOrReturnError(mInitParams.mCASESessionManager != nullptr, CHIP_ERROR_INCORRECT_STATE);
 
     mLastSessionEstablishmentError = CHIP_NO_ERROR;
-    mInitParams.mCASESessionManager->FindOrEstablishSession(nodeId, &mOnConnectedCallback, &mOnConnectionFailureCallback);
+    auto * connectionCallback      = Platform::New<ConnectionCallback>(*this);
+    mInitParams.mCASESessionManager->FindOrEstablishSession(nodeId, connectionCallback->GetOnDeviceConnected(),
+                                                            connectionCallback->GetOnDeviceConnectionFailure());
     if (mLastSessionEstablishmentError == CHIP_ERROR_NO_MEMORY)
     {
         // Release the least recently used entry
-        // TODO: Some reference counting mechanism shall be added the CASESessionManager
-        // so that other session clients don't get accidentally closed.
         ScopedNodeId peerToRemove;
         if (mPendingNotificationMap.FindLRUConnectPeer(peerToRemove) == CHIP_NO_ERROR)
         {
             mPendingNotificationMap.RemoveAllEntriesForNode(peerToRemove);
```

Comment:
How does this help, though?  If we got NO_MEMORY that means the SessionSetup pool is full, no?  How else can we end up here?  And messing with our notification map will not help with that, I don't think; we would have to cancel some session setups, which is what this code _used_ to do when it was working with OperationalDeviceProxy.

This part needs some thought as to how exactly it should work.  Is this what #22173 is about?

Reply:
You are correct.

When I was trying to get this in I was trying to changes as little as possible while fixing https://github.com/project-chip/connectedhomeip/issues/21606, this has been broken before this PR. Part of my thinking is that this should be thought out a little more and addressed in https://github.com/project-chip/connectedhomeip/issues/22173